flink-doris-connector/src/test/java/org/apache/doris/flink/autoci/container/DorisContainerService.java

```diff
@@ -0,0 +1,235 @@
+// under the License.
+
+package org.apache.doris.flink.autoci.container;
+
+import com.google.common.collect.Lists;
+import org.apache.doris.flink.exception.DorisRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.utility.DockerLoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.LockSupport;
+
+public class DorisContainerService implements ContainerService {
+    private static final Logger LOG = LoggerFactory.getLogger(DorisContainerService.class);
+    private static final String DEFAULT_DOCKER_IMAGE = "apache/doris:doris-all-in-one-2.1.0";
+    private static final String DORIS_DOCKER_IMAGE =
+            System.getProperty("image") == null
+                    ? DEFAULT_DOCKER_IMAGE
+                    : System.getProperty("image");
+    private static final String DRIVER_JAR =
+            "https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.16/mysql-connector-java-8.0.16.jar";
+    private static final String JDBC_URL = "jdbc:mysql://%s:9030";
+    private static final String USERNAME = "root";
+    private static final String PASSWORD = "";
+    private final GenericContainer dorisContainer;
+
+    public DorisContainerService() {
+        dorisContainer = createDorisContainer();
+    }
+
+    public GenericContainer createDorisContainer() {
+        LOG.info("Will create doris containers.");
+        GenericContainer container =
+                new GenericContainer<>(DORIS_DOCKER_IMAGE)
+                        .withNetwork(Network.newNetwork())
+                        .withNetworkAliases("DorisContainer")
+                        .withPrivilegedMode(true)
+                        .withLogConsumer(
+                                new Slf4jLogConsumer(
+                                        DockerLoggerFactory.getLogger(DORIS_DOCKER_IMAGE)))
+                        .withExposedPorts(8030, 9030, 8040, 9060);
+
+        container.setPortBindings(
+                Lists.newArrayList(
+                        String.format("%s:%s", "8030", "8030"),
+                        String.format("%s:%s", "9030", "9030"),
+                        String.format("%s:%s", "9060", "9060"),
+                        String.format("%s:%s", "8040", "8040")));
+        return container;
+    }
+
+    public void startContainer() {
```

Comment:
Can this ensure that the entire pipeline reuses the doris cluster?

Reply:
Yes, this container will only be started once.